content/blog/pulumi-loves-python/index.md

```diff
@@ -0,0 +1,100 @@
+---
+title: "Pulumi Loves Python"
+date: 2024-06-28T16:17:38+02:00
+draft: false
+meta_desc: First-class Python support with Poetry, Pythonic input types and built-in type checking.
+meta_image: meta.png
+authors:
+    - julien-poissonnier
+tags:
+    - python
+social:
+    twitter:
+    linkedin:
+
+---
+
+Pulumi has first-class support for Python, and it is one of our most popular languages. We are always working to improve the Python experience, and we are excited to announce some new features that make it even easier to use Python with Pulumi, including support for Poetry, built-in type checking, and more Pythonic input types.
+
+<!--more-->
+
+<!-- expand on the introduction here -->
+
+## Poetry
+
+Pulumi has long had support to create and manage virtual environments using Python's builtin package manager [pip](https://pip.pypa.io/en/stable/). With the [latest release of Pulumi](https://github.com/pulumi/pulumi/releases/tag/v3.121.0), we are excited to announce that we now support [Poetry](https://python-poetry.org) as well. Poetry is a popular Python dependency management tool that allows you to declare your dependencies in a simple and concise way and manage your virtual environment with ease.
+
+When creating a new Pulumi project, the Pulumi CLI will now ask you if you want to use Poetry to manage your dependencies. If you choose to use Poetry, the Pulumi CLI will automatically create a new Poetry project for you and install the necessary dependencies. To opt-in to using Poetry for an existing Pulumi project, set the [`toolchain` runtime option](https://www.pulumi.com/docs/concepts/projects/project-file/#runtime-options) to `poetry` and run `pulumi install`.
```

Comment:
In the next section there's a nice copyable/referenceable  snippet that shows how to set `typechecker`. Any reason not to use a similar snippet here (just to save the user from having to figure out how to do it)?

Reply:
Added